packages/perseus/src/mixins/widget-prop-denylist.js

```diff
@@ -34,3 +36,15 @@ export default [
     "trackInteraction",
     "keypadElement",
 ];
+
+export const removeDenylistProps = (props: $FlowFixMe): $FlowFixMe => {
+    const newProps = {...props};
+    for (const prop of denylist) {
+        if (prop in newProps) {
+            newProps[prop] = undefined;
+        }
+    }
+    return newProps;
+};
```

Comment:
I like that you made this into a utility function to abstract away the repeated logic. I don't think this is a direct port of `omit` though since I think `omit` _deletes_ keys whereas with this logic you're keeping the key and just setting it to undefined. Might be worth adding a test.

``` JS
const a = { foo: "foo", bar: "bar" }
const b = { foo: "foo", bar: "bar" }

a.foo = undefined
delete b.foo

console.log(a); // {foo: undefined, bar: "bar"}
console.log(b); // {bar: "bar"}
```

Reply:
I updated the function and created a test with help from Jeanette to hopefully confirm it is doing what we want it to do.